eclair-core/src/test/scala/fr/acinq/eclair/io/SwitchboardSpec.scala

```diff
@@ -141,6 +144,66 @@ class SwitchboardSpec extends TestKitBaseClass with AnyFunSuiteLike {
     peer.expectMsg(Peer.GetPeerInfo(Some(probe.ref.toTyped)))
   }
 
+  test("track nodes with incoming connections that do not have a channel") {
+    val nodeParams = Alice.nodeParams.copy(peerConnectionConf = Alice.nodeParams.peerConnectionConf.copy(maxNoChannels = 2))
+    val (probe, peer, peerConnection, channel) = (TestProbe(), TestProbe(), TestProbe(), TestProbe())
+    val hasChannelsNodeId1 = randomKey().publicKey
+    val hasChannelsNodeId2 = randomKey().publicKey
+    val unknownNodeId1 = randomKey().publicKey
+    val unknownNodeId2 = randomKey().publicKey
+    val switchboard = TestActorRef(new Switchboard(nodeParams, FakePeerFactory(probe, peer)))
+    switchboard ! Switchboard.Init(Nil)
+
+    // Do not track nodes we connect to.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, randomKey().publicKey, outgoing = true)
+    peer.expectMsgType[Peer.Init]
+
+    // Do not track an incoming connection from a peer we have a channel with.
+    switchboard ! ChannelIdAssigned(channel.ref, hasChannelsNodeId1, randomBytes32(), randomBytes32())
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, hasChannelsNodeId1, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+
+    // We do not yet have channels with these peers, so we track their incoming connections.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, unknownNodeId1, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, unknownNodeId2, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+
+    // Disconnect the oldest tracked peer when an incoming connection from a peer without channels connects.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, randomKey().publicKey, outgoing = false)
+    peer.fishForMessage() {
+      case d: Peer.Disconnect => d.nodeId == unknownNodeId1
+      case _: Peer.Init => false
```

Comment:
This case is unnecessary, `fishForMessage()` take a `PartialFunction`. You can also remove it line 189 below.

Reply:
I use `fishForMessage` because usually `Init` is sent to `peer` but sometimes it can happen that `Peer.Disconnect` is seen first. I'm using it to ignore `Init` and wait for `Peer.Disconnect`. If I remove that line, I get the following error. 
```
assertion failed: fishForMessage() found unexpected message Init(Set())
```

Reply:
Damn, that means it doesn't really accept just a partial function...but then you should ignore *all* other messages, not just `Peer.Init`? It should be `case _ => false`?